content/blog/custom-field-plugins.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Creating Field Plugins
+date: '2020-02-14T07:00:00.000Z'
+author: Kendall Strautman
+draft: false
+consumes:
+  - file: /packages/next-tinacms-json/src/use-local-json-form.ts
+    details: Example shows useLocalJsonForm in use
+  - file: /packages/next-tinacms-json/src/use-json-form.ts
+    details: Example shows useLocalJsonForm in use
+  - file: /packages/@tinacms/form-builder/src/field-plugin.tsx
+    details: Depends on the FieldPlugin interface
+---
+
+In the [previous post](https://tinacms.org/blog/custom-field-components), we learned how to create a custom field component and register it to the sidebar. So let's **go full circle on the topic of custom fields** in TinaCMS. In this short but sweet post we’ll cover how to _turn a plain ol’ field component into a field plugin._
+
+**Getting Started 👏**
+
+To follow along, you should have a custom field component set-up with a Tina form. If not, no worries! Just take a gander at the post on [how to create a custom field component](https://tinacms.org/blog/custom-field-components); it should provide enough working knowledge to get going. In the following examples, I will continue using the [llama-filters](https://github.com/kendallstrautman/llama-filters) demo created in the previous post.
+
+There are **two steps to adding a field plugin to the CMS**. First, we'll define the field component object and register it with the CMS. Next, we'll use the field plugin in a form definition so we can edit content in the sidebar with our fancy custom field plugin.
+
+## 1. Add the Field Plugin to the CMS
+
+To register the custom field as a plugin with the CMS, we’ll need to head to the file where we can access the cms instance. In our Next.js demo, we’ll look at `_app.js`. This it the place where we override the default Next app to wrap `Tina` around each page.
+
+```js
+import React from 'react'
+import App from 'next/app'
+import { Tina, TinaCMS } from 'tinacms'
+import { GitClient } from '@tinacms/git-client'
+/*
+ ** 1. Import the custom field component
+ */
+import RangeInput from '../components/RangeInput'
+
+export default class Site extends App {
+  constructor() {
+    super()
+    this.cms = new TinaCMS({
+      apis: {
+        git: new GitClient('http://localhost:3000/___tina'),
+      },
+      sidebar: {
+        position: 'overlay',
+        hidden: process.env.NODE_ENV === 'production',
+      },
+    })
+
+    /*
+     ** 2. Define the field plugin
+     */
+    this.customFieldPlugin = {
+      name: 'range-input',
+      Component: RangeInput,
+    }
```

Comment:
I might not make the object a property the Site class. There's no real advantage to doing so, and I would bet that people will think it's necessary.

You could just make it a variable scoped to the constructor:

```ts
constructor() {
  // ...
  let rangeFieldPlugin = {
    name: 'range-input',
    Component: RangeInput
  }
  // ...
}
```

or you could make it a module-scoped variable:

```
let rangeFieldPlugin = {
  name: 'range-input',
  Component: RangeInput
}

export default class Site extends App {
  // ...
```

Personally, I prefer the constructor scope for this blog post because it maintains the order for your instructions.